app/helpers/payments_helper.rb

```diff
@@ -4,32 +4,63 @@ module PaymentsHelper
   # set based on whether or not the date has expired and a tooltip that explains it.
   #
   # @param entity [User | Object]- the entity that provides the expiration date. If a User,
-  # must respond to :membership_expire_date; else must respond to :branding_expire_date
+  # must respond to :entity_expire_date
   #
   # @return [String] - the HTML <span> string
   def expire_date_label_and_value(entity)
-    if entity.is_a? User
-      expire_date = entity.membership_expire_date
-      expire_after_tooltip_title = "#{t('users.show.membership_expire_date_tooltip')}"
+
+    expire_date = entity_expire_date(entity)
+    t_scope = entity.is_a?(User) ? 'users' : 'companies' # TODO - should use polymorphism to handle this
+
+    expire_after_tooltip_title = t("#{t_scope}.show.term_expire_date_tooltip")
+    expire_label = t("#{t_scope}.show.term_paid_through")
+
+    if expire_date
+      tag.div do
+        concat tag.span "#{expire_label}: ", class: 'standard-label'
+        concat tag.span "#{expire_date}", class: payment_should_be_made_class(entity)
+        concat ' '
+        concat fas_tooltip(expire_after_tooltip_title)
+      end
     else
-      expire_date = entity.branding_expire_date # Company
-      expire_after_tooltip_title = "#{t('companies.show.branding_fee_expire_date_tooltip')}"
+      field_or_none(expire_label, t('none_t'), label_class: 'standard-label')
     end
+  end
 
-    unless expire_date
-      return field_or_none("#{t('activerecord.attributes.payment.expire_date')}",
-                           "#{t('none_t')}", label_class: 'standard-label')
+
+  # @param entity [User or Company] - the entity with a possible membership expiration date
+  # @return [nil | date] - return nil if there is no expiration date (e.g. not a member), else
+  # the Date that the current membership term expires
+  # TODO - this is a smell (checking the class): polymorphism should be used instead. Default implementation should be in a parent class for User and Company
+  def entity_expire_date(entity = nil)
```

Comment:
Not sure I agree with this statement.  User and Company are quite distinct, and making them subclasses of a parent (Payor?) class would be a mistake, IMO.  The two classes would be subclassed based on a common attribute, not based on being a "type of" or specific examples of the parent class.

Reply:
Agree that User and Company have a lot of differences.

But they do definitely share the common behavior of making and handling _payments_.
This behavior is what I think of as a `Payor`   Refactoring out this behavior, and then refactoring out the different `Payments` is what we should discuss and work through.